### PR TITLE
 Changes due to the implicit index creation

### PR DESCRIPTION
**Pull request by curquiza**
Related to https://github.com/meilisearch/integration-guides/issues/48

Changes:
- New method `$client->index($uid)`: replaces `$client->getIndex($uid)`. Does not do any HTTP call but only returns an `Index` instance.
- New attribute in the `Index` class: `primaryKey`
- New method `$index->fetchRawInfo()`: returns an array of index
- New method `$index->fetchInfo()`: returns an `Index` instance

Breaking changes:
- `$client->getIndex($uid)` now does an HTTP call. Use `$client->index($uid)` instead.
- `$client->updateIndex` returns an `Index` instance.
- `$index->getPrimaryKey()` does not do any HTTP call, but only return the `primaryKey` attribute. Use `$index->fetchPrimaryKey()` if you want the same behavior as the old `$index->getPrimaryKey()`
- `$index->show()` is removed and replaced by `$index->fetchInfo()` (returns an instance) and `$index->fetchRawInfo()` (returns an array)
- `$client->showIndex($uid)` is removed. Use `$client->getIndex($uid)` (returns an instance) or `$client->index($uid)->fetchRawInfo()` (returns an array) instead.

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -4,25 +4,25 @@
 # You can read more on https://github.com/meilisearch/documentation/tree/master/.vuepress/code-samples
 ---
 get_one_index_1: |-
-  $client->getIndex('movies');
+  $client->index('movies')->getRaw();
```
Comment (eskombro) [inline .code-samples.meilisearch.yaml:7]: Shouldn't this use `fetchInfo()` instead of `getRaw()`, returning an Index instead of an http response?

**Review by eskombro — APPROVED** [review]
Great implementation @curquiza ! TY

🎉

File: .code-samples.meilisearch.yaml (hunk 1 above)
Comment (curquiza) [reply .code-samples.meilisearch.yaml:7]: we chose to let the raw option for the moment -> https://github.com/meilisearch/meilisearch-js/pull/715/files#r539555931

Comment (eskombro) [reply .code-samples.meilisearch.yaml:7]: Ok, then everything LGTM 🎉

**Review by eskombro — APPROVED** [review]
👌🏽

**Comment by curquiza** [comment]
Bors merge

**Pull request merged**